cocos/2d/CCNode.h

```diff
@@ -914,6 +927,25 @@ class CC_DLL Node : public Ref
      */
     virtual void sortAllChildren();
 
+    /**
+    * Sorts helper function
+    *
+    */
+    template<typename _T> inline
+    static void sortNodes(cocos2d::Vector<_T*>& nodes)
+    {
+        static_assert(std::is_base_of<Node, _T>::value, "Node::sortNodes: Only accept derived of Node!");
+#if CC_64BITS
+        std::sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
+            return (n1->_localZOrderAndArrival < n2->_localZOrderAndArrival);
+        });
+#else
+        std::stable_sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
+            return n1->_localZOrder < n2->_localZOrder;
+        });
+#endif
+    }
+
```

Comment:
@halx99 i think they have different result on 32bit and 64bit, for example, the codes are
```c++
this->addChild(a, 1);
this->addChild(b, 1); // b will be drawn after a

// on 64bit, a will be drawn after b since its oder arrival is bigger than b
// on 32bit, a will be drawn before b since on 32 bit only compares localZOrder
this->reorder(a, 1);
```

Reply:
Maybe it's missing the orderOfArrival on 32bit.

Reply:
The lambda on 32bit doesn't use orderOfArrival at all.

Reply:
Yes, so the code need change to

        static_assert(std::is_base_of<Node, _T>::value, "Node::sortNodes: Only accept derived of Node!");
#if CC_64BITS
        std::sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
            return (n1->_localZOrderAndArrival < n2->_localZOrderAndArrival);
        });
#else
        std::stable_sort(std::begin(nodes), std::end(nodes), [](_T* n1, _T* n2) {
            return n1->_localZOrder < n2->_localZOrder || (n1->_localZOrder == n2->_localZOrder && n1->_getOrderOfArrival() < n2->_getOrderOfArrival() );
        });
#endif

Reply:
And on 32bit no need to use std::stable_sort since use orderOfArrival, what do you think abou it.

Reply:
Let's talk at here: https://github.com/cocos2d/cocos2d-x/pull/18233

Reply:
Yep, the codes looks good, but there is not `Node::_getOrderOfArrival()`, if we retrieve it from `_localZOrderAndArrival`, then the performance will be affected.